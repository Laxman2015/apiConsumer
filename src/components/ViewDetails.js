import React, { Fragment } from "react";
import { useSelector } from "react-redux";
import { useHistory } from "react-router";
import styles from "./Details.module.scss";
import { Star } from "./Star";

const ViewDetails = (props) => {
  const id = props.match ? props.match.params.id : 1;
  const { bookLists } = useSelector((state) => state);
  const product = bookLists.find((book) => book.id == id);

  const history = useHistory();

  const backToHome = () => history.push('/');

  const starRender = (num) => {
    let rating = [];
    for (let i = 0; i < num; i++) {
      rating.push(<Star width={"12px"} />);
    }
    return rating;
  };
  return (
    <Fragment>
        <h1 className={styles.title}>Book Details</h1>
      <div className={styles.cartItem}>
        <div className={styles.imgWidth}>
          <img
            alt={product.name}
            className={styles.bookImage}
            src={product.url}
            className="img-fluid d-block"
          />
        </div>
        <div className={styles.prodInfo}>
          <h3 className="mb-1">BookID: {product.id}</h3>
          <h3 className="mb-1">Name: {product.name}</h3>
          <h4 className="mb-1">Raiting: {starRender(product.star)}</h4>
          <h4 className="mb-1">Price: {product.price} </h4>
        </div>
      </div>
      <div className={styles.backToHome}>
        <button className={styles.btnPrimary} onClick={() => backToHome()}>Back To Home</button>
      </div>
    </Fragment>
  );
};

export default ViewDetails;
