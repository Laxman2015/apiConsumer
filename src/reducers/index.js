import { combineReducers } from "redux";
import BookReducer from "./reducers";

// const rootReducer = combineReducers({
//   BookReducer,
// });

export default BookReducer;
