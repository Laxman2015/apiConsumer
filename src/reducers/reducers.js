import initialState from './initialState';

export default function BookReducer(state = initialState, action) {
  switch (action.type) {
    case "UPDATE_BOOKLISTS":
      return {
        ...state,
        bookLists: action.payload
      };
    
    default:
      return state;
  }
}
