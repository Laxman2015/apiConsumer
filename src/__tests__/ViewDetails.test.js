import React from "react";
import { mount, shallow } from "enzyme";

import ViewDetails from "../components/ViewDetails";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../reducers";
import { BrowserRouter } from "react-router-dom";

const store = createStore(reducers);

const props = {
  props: {
    match: {
      params: {
        id: 1,
      },
    },
  },
};
const mockConfigState = {
  bookLists: [
    {
      id: 1,
      name: "Book1",
      price: 10,
      star: 3,
      url: "https://test.com",
    },
    {
      id: 2,
      name: "Book2",
      price: 15,
      star: 5,
      url: "https://test2.com",
    },
    {
      id: 3,
      name: "Book3",
      price: 20,
      star: 3,
      url: "https://test3.com",
    },
  ],
};

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest
    .fn()
    .mockReturnValueOnce(mockConfigState)
    .mockReturnValueOnce(mockConfigState),
}));

describe("ViewDetails Component", () => {
  mockConfigState.getState = jest.fn();
  it("renders ViewDetails", () => {
    const wrapper = shallow(
      <Provider store={mockConfigState}>
        <BrowserRouter>
          <ViewDetails {...props} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").length).toBe(0);
    expect(wrapper.find(ViewDetails).length).toBe(1);
  });
});
