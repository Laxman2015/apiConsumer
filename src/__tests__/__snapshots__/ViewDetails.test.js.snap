// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDetails Component renders ViewDetails 1`] = `ShallowWrapper {}`;
